supybot_fedora/plugin.py

```diff
@@ -195,46 +213,104 @@ def __init__(self, irc):
+                    "/user/list", req_params={"search": "*"}, auth=True, timeout=240
+                )
+                users = request["people"] + request["unapproved_people"]
+                del request
+            except AuthError:
+                self.log.info("Error Authorizing to FAS")
+                users = []
+
+            self.log.info("Caching necessary user data")
+            self.users = {}
+            self.faslist = {}
+            self.nickmap = {}
+            for user in users:
+                name = user["username"]
+                self.users[name] = {}
+                self.users[name]["id"] = user["id"]
+                key = " ".join(
+                    [
+                        user["username"],
+                        user["email"] or "",
+                        user["human_name"] or "",
+                        user["ircnick"] or "",
+                    ]
+                )
+                key = key.lower()
+                value = "%s '%s' <%s>" % (
                     user["username"],
-                    user["email"] or "",
                     user["human_name"] or "",
-                    user["ircnick"] or "",
-                ]
-            )
-            key = key.lower()
-            value = "%s '%s' <%s>" % (
-                user["username"],
-                user["human_name"] or "",
-                user["email"] or "",
-            )
-            self.faslist[key] = value
-            if user["ircnick"]:
-                self.nickmap[user["ircnick"]] = name
+                    user["email"] or "",
+                )
+                self.faslist[key] = value
+                if user["ircnick"]:
+                    self.nickmap[user["ircnick"]] = name
+
+            socket.setdefaulttimeout(timeout)
 
-        socket.setdefaulttimeout(timeout)
+    def _get_person_by_username(self, irc, username):
+        """looks up a user by the username"""
+        if self.registryValue("use_fasjson"):
+            try:
+                person = self.fasjsonclient.get_user(username=username).result
+            except fasjson_client.errors.APIError as e:
+                if e.code == 404:
+                    irc.reply(f"Sorry, but user '{username}' does not exist")
+                    return
+                else:
+                    irc.reply("Something blew up, please try again")
+                    self.log.error(e)
+                    return
+        else:
+            try:
+                person = self.fasclient.person_by_username(username)
+                # convert to the newer FASJSON way for now
+                if person.get("email"):
+                    person["emails"] = [person["email"]]
+            except Exception as e:
+                irc.reply("Something blew up, please try again")
+                self.log.error(e)
+            if not person:
+                irc.reply(f"Sorry, but user '{username}' does not exist")
+
+        return person or None
```

Comment:
What falsy value do you anticipate `person` to be? I'm not sure which case you're trying to cover here. Also, all the calls to this functions are followed by a `if not person` so I don't think the `or None` is useful here.

Reply:
yeah, the or None here was just for the fasclient case. I was trying to avoid changing that logic at all, but it would probably be better to just put the returns in like:

```diff
        else:
            try:
                person = self.fasclient.person_by_username(username)
                # convert to the newer FASJSON way for now
                if person.get("email"):
                    person["emails"] = [person["email"]]
            except Exception as e:
                irc.reply("Something blew up, please try again")
                self.log.error(e)
+              return
            if not person:
                irc.reply(f"Sorry, but user '{username}' does not exist")
+              return

-        return person or None
+        return person

```